test/ensembl/ensembl.uploadFile.test.js

```diff
@@ -0,0 +1,255 @@
+            feature: {
+                biotype: 'transcript', source: 'ensembl', sourceId: 'ENST00000680887', sourceIdVersion: null,
+            },
+        },
+        {
+            '@rid': '#138:3',
+            edges: [],
+            feature: {
+                biotype: 'gene', name: 'BRCA2', source: 'entrez gene', sourceId: '675',
+            },
+        },
+        {
+            '@rid': '#138:1',
+            edges: [
+                { class: 'crossreferenceof', in: '#138:3' },
+            ],
+            feature: { biotype: 'gene', name: 'BRCA2', source: 'hgnc' },
+        },
+        {
+            '@rid': '#140:1',
+            edges: [
+                { class: 'deprecatedby', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'FACD', source: 'hgnc' },
+        },
+        {
+            '@rid': '#137:2',
+            edges: [
+                { class: 'deprecatedby', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'FANCD', source: 'hgnc' },
+        },
+        {
+            '@rid': '#139:1',
+            edges: [
+                { class: 'deprecatedby', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'FANCD1', source: 'hgnc' },
+        },
+        {
+            '@rid': '#138:2',
+            edges: [
+                { class: 'aliasof', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'FAD', source: 'hgnc' },
+        },
+        {
+            '@rid': '#139:2',
+            edges: [
+                { class: 'aliasof', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'FAD1', source: 'hgnc' },
+        },
+        {
+            '@rid': '#140:2',
+            edges: [
+                { class: 'aliasof', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'BRCC2', source: 'hgnc' },
+        },
+        {
+            '@rid': '#137:3',
+            edges: [
+                { class: 'aliasof', in: '#138:1' },
+            ],
+            feature: { biotype: 'gene', name: 'XRCC11', source: 'hgnc' },
+        },
+        {
+            '@rid': '#137:0',
+            edges: [],
+            feature: {
+                biotype: 'transcript', source: 'refseq', sourceId: 'NM_000059', sourceIdVersion: null,
+            },
+
+        },
+    ].map(el => Object.assign(el, {
+        // Custom method - Returns a string representation of the feature
+        toString: () => ('name' in el.feature
+            ? `${el.feature.biotype} Feature ${el.feature.name} from ${el.feature.source}`
+            : `${el.feature.biotype} Feature ${el.feature.sourceId} from ${el.feature.source}`),
+    })).map(el => Object.assign(el, {
+        // Custom method - Returns the source's RID from mockDataset
+        getSourceRid: () => ((conn.request).mock.results.filter(
+            (a) => a.value.result.name === el.feature.source,
+        )[0].value.result['@rid']),
```

Comment:
so far the test dataset only has 1 result, would this be a problem when it is not of length 1?

Reply:
When testing the loader for adding the right Feature objects to the database, we use the getSourceRid() method to make sure the Feature has been given the right Source RID as Source property value: https://github.com/bcgsc/pori_graphkb_loader/blob/5a6a7bba4e8d37071f0788c24d496826f37477a9/test/ensembl/ensembl.uploadFile.test.js#L236

This is why every object of the 'expected' list has been given the same method allowing them to convert their source from name to RID. The correct RID is obtained from returned values from the Jest mock object. If more than one result returned the same name, then it should also return the same RID, so we can juste grab the first returned result : https://github.com/bcgsc/pori_graphkb_loader/blob/5a6a7bba4e8d37071f0788c24d496826f37477a9/test/ensembl/ensembl.uploadFile.test.js#L216

Same for the edges here:
https://github.com/bcgsc/pori_graphkb_loader/blob/5a6a7bba4e8d37071f0788c24d496826f37477a9/test/ensembl/ensembl.uploadFile.test.js#L249

If we want to make sure that the name of a source is not also the name of an other object returned by the mock dataset, then we could also add a @class filter. I'll take care of adding that; thanks @kttkjl for the comment.

Reply:
Ohh I see, thanks for clearing that up!